Fix: use native language for desciptions of a couple of predefined MUDS

SlySven opened the pull request:
Following the inclusion of "Mud.ren" with its Chinese description text (in https://github.com/Mudlet/Mudlet/pull/6140), I felt it was appropriate to put in the native language texts for a Spanish and an Italian MUD who have already provided us with the text to use but which we hadn't so far. This will address points which I made about this in: https://github.com/Mudlet/Mudlet/pull/1505 and https://github.com/Mudlet/Mudlet/pull/2502#discussion_r273176458 respectively...

I have cleaned up the texts involved for all the descriptions so that they are all wrapped in our `qsl(`...`)` wrapper and the
host URLs that are used to identify which one to use are all wrapped in `QLatin1String(`...`)` as they should all be ASCII (as they are all OS compatible file names as well) and there is a `QLatin1String::operator==()` method for them.

In passing I have also "fixed" the "website" entries for the Spanish MUD as it is appropriate for them to be in Spanish rather than English as they are not URLs but HTTP-like `<a href="xxxx">`...`</a>` entries.

I propose to follow this PR with another one to move the description data to the same place as all the other data about the predefined MUDs (the container defined in the `./src/mudlet.h` file: `(inline static const OrderedMap<QString, GameDetails>) mudlet::scmDefaultGames`)

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Fix: use native language for desciptions of a couple of predefined MUDS

Following the inclusion of "Mud.ren" with its Chinese description text
(in https://github.com/Mudlet/Mudlet/pull/6140),
I felt it was appropriate to put in the native language texts for a
Spanish and an Italian MUD who have already provided us with the text
to use but which we hadn't so far. This will address points which I
made about this in:
https://github.com/Mudlet/Mudlet/pull/1505
and
https://github.com/Mudlet/Mudlet/pull/2502#discussion_r273176458
respectively...

I have cleaned up the texts involved for all the descriptions so
that they are all wrapped in our `qsl(`...`)` wrapper and the
host URLs that are used to identify which one to use are all wrapped
in `QLatin1String(`...`)` as they should all be ASCII (as they are
all OS compatible file names as well) and there is a
`QLatin1String::operator==()` method for them.

In passing I have also "fixed" the "website" entries for the Spanish
MUD as it is appropriate for them to be in Spanish rather than English
as they are not URLs but HTTP-like `<a href="xxxx">`...``</a>` entries.

I propose to follow this PR with another one to move the description
data to the same place as all the other data about the predefined MUDs
(the container defined in the `./src/mudlet.h` file:
`(inline static const OrderedMap<QString, GameDetails>)
                                               mudlet::scmDefaultGames`)

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

Kebap commented on src/dlgConnectionProfiles.cpp:
```diff
@@ -716,166 +716,311 @@ QPair<bool, QString> dlgConnectionProfiles::writeProfileData(const QString& prof
-    } else if (hostUrl == QLatin1String("3k.org")) {
+        // clang-format off
+        // wrap before the last word on a line is here so that the source text
+        // is no more than 80 characters excluding the indent for readbility in
```
```suggestion
        // is no more than 80 characters (excluding the indent) for readbility in
```

Kebap commented on src/dlgConnectionProfiles.cpp:
```diff
@@ -716,166 +716,311 @@ QPair<bool, QString> dlgConnectionProfiles::writeProfileData(const QString& prof
+                   "are always willing to answer questions and to help out however necessary. Best "
+                   "of all, playing the Realms of Despair is totally FREE!");
+    }
+    if (hostUrl == QLatin1String("zombiemud.org")) {
```
I liked "else if" better here, as this somewhat contradicts the previous and needn't be checked if that was true.

Kebap replied:
Oh wait, if the "if" block actually ended in "return", then we can indeed drop the "else" here.

SlySven replied:
This PR removes the "else-after-return" anti-patterns in this method that some *linters* - such as `clang-tidy` moan about, see: https://clang.llvm.org/extra/clang-tidy/checks/readability/else-after-return.html . If the previous if test is true, execution never reaches here as the return short-circuits the remainder of the method.

Kebap commented on src/dlgConnectionProfiles.cpp:
```diff
@@ -716,166 +716,311 @@ QPair<bool, QString> dlgConnectionProfiles::writeProfileData(const QString& prof
+                   "multi-player games ever!");
+    }
+    if (hostUrl == QLatin1String("reinosdeleyenda.es")) {
+// English translation (was used prior to switch to native Spanish of the MUD)
```
"was used prior" etc. Could be dropped, unnecessary as part of git history.
Good idea to keep the English translation here nonetheless.
Would love to see a future PR to toggle display language here as well.

SlySven replied:
You want just this line or the whole English translation removed?

SlySven replied:
> Would love to see a future PR to toggle display language here as well.

There is no point IMVHO - someone is not going to want to read about a MUD that is both described AND played in a language that they cannot comprehend - if anything such a text will guide them away from such predefined MUDs...

SlySven replied:
Also, the first line comment does explain why the remaining lines - though being valid code/text - IS commented out - it is one way to retain the English version of the text... 🤷‍♂️

Kebap replied:
> You want just this line or the whole English translation removed?

The text in brackets only.

Kebap commented on src/dlgConnectionProfiles.cpp:
```diff
@@ -716,166 +716,311 @@ QPair<bool, QString> dlgConnectionProfiles::writeProfileData(const QString& prof
+                   "提剑跨骑挥鬼雨，白骨如山鸟惊飞。\n"
+                   "尘事如潮人如水，只叹江湖几人回。"
+                   "\n\n"
+                   "中文开源武侠MUD炎黄群侠传，游戏包括25大门派和5大世家，正邪只在一念间；近千门武学等你学习，上百种任务随你体验；让自己成为一代宗师，江湖笑，恩怨了。");
```
Please ask for English translation to deposit here as well.

SlySven replied:
I had no idea what the translation was, I mentioned what I got from Google's translation when @vadi2 put that text in - as you picked up in https://github.com/Mudlet/Mudlet/pull/6140#issuecomment-1179811763 - OTOH there is no absolute reason why there will be a precisely equivalent English text - it isn't an English MUD. 🤷‍♂️

Kebap replied:
They added a brief English description like Kung fu game, would you include it here as first paragraph?

vadi2 replied:
The game author is really uncomfortable with an English description - what would it add?

Kebap replied:
Here is a brief one-line description they gave: https://github.com/Mudlet/Mudlet/pull/6140#issuecomment-1184121811

I think it helps English players gather the gist of what happens in Mudlet still, not be dazzled by lots foreign language texts alone.

SlySven replied:
TBH By definition a foreign language text will put off English players - which is actually appropriate IMVHO - and why should we be treating **English** players differently than say German or Russian ones? All in all I now think it would be reasonable to merge this PR in as it is.

\<aside>Perhaps we should be seeking out suitable description texts for all the predefined MUDs that do not currently have anything - for instance this would help to avoid *English* only speakers from clicking on Avalon without spotting that it is a German MUD!\</aside>

Kebap reviewed (commented):
Good idea to change the text source to maximum length like the display in the app widget will do as well. Makes it easier to compare in the future. Makes it more difficult to compare this PR but hey.

SlySven commented:
> Good idea to change the text source to maximum length like the display in the app widget will do as well. Makes it easier to compare in the future. Makes it more difficult to compare this PR but hey.

"... like the display in the app widget will do as well" not exactly - I just worked to an 80 character width so that one could read whole lines at a time (assuming that 80 characters was a reasonable default) - you can resize the Connect Profile dialogue quite a bit and the text will reflow/rewrap to fit that widget in the app.

SlySven committed:
Update src/dlgConnectionProfiles.cpp

Revised a comment slightly...

Co-authored-by: Kebap <kebap_spam@gmx.net>

SlySven commented on src/dlgConnectionProfiles.cpp:
```diff
@@ -716,166 +716,311 @@ QPair<bool, QString> dlgConnectionProfiles::writeProfileData(const QString& prof
+                   "multi-player games ever!");
+    }
+    if (hostUrl == QLatin1String("reinosdeleyenda.es")) {
+// English translation (was used prior to switch to native Spanish of the MUD)
+//        return qsl("The oldest Spanish free mud with more than 20 years of running history."
+//                   "\n\n"
+//                   "Reinos de Leyenda takes place in the ever changing world of Eirea, ravaged by the "
+//                   "mischiefs of the gods after more than a thousand years of contempt and hideous war "
+//                   "amongst their zealous mortal pawns."
+//                   "\n\n"
+//                   "History is written on a day per day basis, taking into consideration the players' "
+//                   "choices to decide the irreversible aftermath of this everlasting struggle."
+//                   "\n\n"
+//                   "This is a PvP MUD which allows the player to set how high are the stakes: the more "
+//                   "you risk losing upon death, the more glory to be earned by your heroism. RP, while "
+//                   "not enforced, is rewarded with non-PvP oriented perks and unique treasure."
+//                   "\n\n"
+//                   "A powerful character customization system allows you to choose your deity –or "
+//                   "fully disregard the gods– and join one of the player-run realms that govern the "
+//                   "land to explore a breathing world, delve into the secrets of the oceans, shape "
+//                   "your legacy, craft forgotten marvels for you –or your allies– and fight for faith, "
+//                   "glory or coin.");
```
```suggestion
/* English translation, provided by Game:
 *                   "The oldest Spanish free mud with more than 20 years of running history."
 *                   "\n\n"
 *                   "Reinos de Leyenda takes place in the ever changing world of Eirea, ravaged by the "
 *                   "mischiefs of the gods after more than a thousand years of contempt and hideous war "
 *                   "amongst their zealous mortal pawns."
 *                   "\n\n"
 *                   "History is written on a day per day basis, taking into consideration the players' "
 *                   "choices to decide the irreversible aftermath of this everlasting struggle."
 *                   "\n\n"
 *                   "This is a PvP MUD which allows the player to set how high are the stakes: the more "
 *                   "you risk losing upon death, the more glory to be earned by your heroism. RP, while "
 *                   "not enforced, is rewarded with non-PvP oriented perks and unique treasure."
 *                   "\n\n"
 *                   "A powerful character customization system allows you to choose your deity –or "
 *                   "fully disregard the gods– and join one of the player-run realms that govern the "
 *                   "land to explore a breathing world, delve into the secrets of the oceans, shape "
 *                   "your legacy, craft forgotten marvels for you –or your allies– and fight for faith, "
 *                   "glory or coin.");
 */
```

Kebap replied:
We call them "games" not "MUD" but otherwise this is an improvement indeed I approve.

SlySven commented on src/dlgConnectionProfiles.cpp:
```diff
@@ -716,166 +716,311 @@ QPair<bool, QString> dlgConnectionProfiles::writeProfileData(const QString& prof
+                   "aliados- y luchar por fe, gloria o dinero.");
+    }
+    if (hostUrl == QLatin1String("mud.clessidra.it")) {
+// English translation (was used prior to switch to native Italian of the MUD)
+//        return qsl("Clessidra is the first all italian MUD ever created! On Clessidra you may find "
+//                   "only original Areas, all in italian! Many features make Clessidra one of the best, "
+//                   "or the best MUD in Italy: Advanced travel mode, fight one to one versus your "
+//                   "friend, or enemy, The Arena and its fight, the Mortal Challenge, the intelligent "
+//                   "MOBs and their Quest and fighting style, a random automatic mission assignament "
+//                   "and for you and your friends you must try the advanced Clan system that allows "
+//                   "wars and conquest. A mercenary system to help playing when few players are online, "
+//                   "a crafting system to create special object and a graphical user interface to help "
+//                   "newbie and expert players have a better experience. A MUD that evolves with new "
+//                   "challenge, new rules, new skills!");
```
```suggestion
/* English translation, provided by Game:
 *                   "Clessidra is the first all italian MUD ever created! On Clessidra you may find "
 *                   "only original Areas, all in italian! Many features make Clessidra one of the best, "
 *                   "or the best MUD in Italy: Advanced travel mode, fight one to one versus your "
 *                   "friend, or enemy, The Arena and its fight, the Mortal Challenge, the intelligent "
 *                   "MOBs and their Quest and fighting style, a random automatic mission assignament "
 *                   "and for you and your friends you must try the advanced Clan system that allows "
 *                   "wars and conquest. A mercenary system to help playing when few players are online, "
 *                   "a crafting system to create special object and a graphical user interface to help "
 *                   "newbie and expert players have a better experience. A MUD that evolves with new "
 *                   "challenge, new rules, new skills!");
 */
```

Kebap replied:
see above

SlySven commented on src/dlgConnectionProfiles.cpp:
```diff
@@ -716,166 +716,311 @@ QPair<bool, QString> dlgConnectionProfiles::writeProfileData(const QString& prof
-    } else {
-        return readProfileData(profile_name, qsl("description"));
-    }
+                   "皇图霸业谈笑中，不胜人生一场醉。\n"
```
```suggestion
/* English translation (courtesy of Google NOT the originator):
 *                   "The world is out of my generation."
 *                   "Emperor Tu Baye talked about laughter, it was drunk in life."
 *                   "The sword rides on the ride, and the white bone is like a mountain bird to fly."
 *                   "The dust is like a tide, and only a few people in the rivers and lakes."
 *                   "\n\n"
 *                   "Chinese open source martial arts MUD Yan Huang Qunxia biography, the game includes 25 masters and 5 major families. Laughing, grudge."
 *\
                   "皇图霸业谈笑中，不胜人生一场醉。\n"
```

Kebap replied:
Not sure if we should store this preliminary google translation if we both can't even comprehend.
Then again, we didn't receive any translation by the originators so this is the best we operate on.

SlySven reviewed (commented):
@Kebap I'm proposing revisions of the presentation of the English text for the two European non-English MUDs and inserted what Google makes of the Chinese one - will this be sufficient to meet with your approval if I commit them?

SlySven committed:
Apply suggestions from code review

Taking into account @Kebap 's point that we call them "(Gg)ame(s)" rather than MUD(s)...

SlySven commented on src/dlgConnectionProfiles.cpp:
```diff
@@ -716,166 +716,321 @@ QPair<bool, QString> dlgConnectionProfiles::writeProfileData(const QString& prof
+ *                   "The dust is like a tide, and only a few people in the rivers and lakes."
+ *                   "\n\n"
+ *                   "Chinese open source martial arts MUD Yan Huang Qunxia biography, the game includes 25 masters and 5 major families. Laughing, grudge."
+ *\
```
Oops! I didn't end that multi-line comment correctly - no wonder CodeFactor through a wobbly on it (and the CI processed failed)!
```suggestion
 */
```
🙄

SlySven reviewed (commented):
Needed to fix an error I introduced in last commit!

SlySven committed:
Update src/dlgConnectionProfiles.cpp

Fix error in last commit.

vadi2 reviewed (approved):
LGTM, let's wait for @Kebap's tick

Kebap commented on src/dlgConnectionProfiles.cpp:
```diff
@@ -716,166 +716,321 @@ QPair<bool, QString> dlgConnectionProfiles::writeProfileData(const QString& prof
+ *                   "your legacy, craft forgotten marvels for you –or your allies– and fight for faith, "
+ *                   "glory or coin.");
+ */
+        return qsl("El mud Español gratis con más de 20 años de historia."
```
Please capitalize MUD like everywhere

```suggestion
        return qsl("El MUD Español gratis con más de 20 años de historia."
```

SlySven replied:
Given that the translation of "Multi User Dungeon" in Spanish appears (according to Google Translate) to be "Dungeon de múltiples usuarios" perhaps we should use "DdMU" - I think we need to get further clarification as to what the correct term ***is*** - and that can follow in a further PR if we get it - remember that that text did come from someone who is associated with it IIRC!

https://www.spanishdict.com/guide/capitalization-in-spanish gives some suggestions in the area of capitalisation but nothing that I can see explicitly describes this case.

Kebap replied:
It's spelled MUD further down in their text, so I was aiming at some professional alignment here

Kebap reviewed (approved):
Fair enough
